hack/bootstrap-ginkgo.sh

```diff
@@ -4,12 +4,16 @@ set -e
 
 source $(dirname "$0")/common.sh
 
-_out/tests/ginkgo version
+FOLDERS="${KUBEVIRT_DIR}/staging/src/kubevirt.io/ ${KUBEVIRT_DIR}/pkg/ ${KUBEVIRT_DIR}/cmd/ ${KUBEVIRT_DIR}/tests/framework/"
+
+ginkgobin=$(realpath _out/tests/ginkgo)
+$ginkgobin version
 # Find every folder containing tests
-for dir in $(find ${KUBEVIRT_DIR}/pkg/ -type f -name '*_test.go' -printf '%h\n' | sort -u); do
+for dir in $(find ${FOLDERS} -type f -name '*_test.go' -printf '%h\n' | sort -u); do
     # If there is no file ending with _suite_test.go, bootstrap ginkgo
     SUITE_FILE=$(find $dir -maxdepth 1 -type f -name '*_suite_test.go')
     if [ -z "$SUITE_FILE" ]; then
-        (cd $dir && ginkgo bootstrap || :)
+        echo "Missing test suite entrypoint; attempt to create one automatically"
+        (cd $dir && $ginkgobin bootstrap)
```

Comment:
I want to verify my understanding: The purpose of this command is to dirty the git tree and fail the make generate lane, right?
If so, could we just create any file (even an empty one) and not depend on Ginkgo's utility?

And in case this assumption is wrong, isn't the file we use has a custom function call?
https://github.com/kubevirt/kubevirt/blob/1a50ec56c184c25fcb4e88777e3dd3e4c184daee/pkg/network/multus/multus_suite_test.go#L29

Reply:
> I want to verify my understanding: The purpose of this command is to dirty the git tree and fail the make generate lane, right?

Yes

> If so, could we just create any file (even an empty one) and not depend on Ginkgo's utility?
> 
> And in case this assumption is wrong, isn't the file we use has a custom function call?

Interesting idea. This would force the developer to spend some time understanding what needs to go where, so I am not sure. It would be great if we could rely on the ginkgo CLI and not have to reinvent things

Reply:
I would assume it's the generated one by ginkgo, but maybe we just have more requirements

Reply:
It does a bit more:
https://github.com/kubevirt/kubevirt/blob/d096f82c3382cfaeda2467deeb711e20a5c59b4a/staging/src/kubevirt.io/client-go/testutils/setup.go#L28

I wonder if this logic is still relevant today.

Reply:
Yeah not sure tbh.. I wouldn't want to enforce this way of initializing ^ and would also not want to have to explain to developers that they should be creating a suite file manually